src/nomad_simulations/schema_packages/properties/energies.py

```diff
@@ -20,33 +20,65 @@
 
 import numpy as np
 
-from nomad.metainfo import Quantity
+from nomad.metainfo import Quantity, Section, Context, SubSection
 
 if TYPE_CHECKING:
     from nomad.metainfo import Section, Context
     from nomad.datamodel.datamodel import EntryArchive
     from structlog.stdlib import BoundLogger
 
-from nomad_simulations.schema_packages.physical_property import PhysicalProperty
+from nomad_simulations.schema_packages.physical_property import (
+    PhysicalProperty,
+    PropertyContribution,
+)
 
+##################
+# Abstract classes
+##################
 
-class FermiLevel(PhysicalProperty):
+
+class BaseEnergy(PhysicalProperty):
     """
-    Energy required to add or extract a charge from a material at zero temperature. It can be also defined as the chemical potential at zero temperature.
+    Abstract physical property section describing some energy of a (sub)system.
     """
 
-    # ! implement `iri` and `rank` as part of `m_def = Section()`
-
-    iri = 'http://fairmat-nfdi.eu/taxonomy/FermiLevel'
-
     value = Quantity(
         type=np.float64,
         unit='joule',
         description="""
-        The value of the Fermi level.
+        The value of the energy.
         """,
     )
 
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class EnergyContribution(BaseEnergy, PropertyContribution):
+    """
+    Abstract physical property section linking a property contribution to a contribution
+    from some method.
+    """
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
```

Comment:
bear in mind that double inheritance implies that you need to specify here the `super()` normalization.

Reply:
what do you mean exactly? I have to choose one parent class for the normalization tree? How do I do this?

Reply:
So if you have a class which double inherits from other 2, then the `super()` calls makes a specific call to those classes functions, executing them in parallel (it is called [method resolution order](https://www.geeksforgeeks.org/method-resolution-order-in-python-inheritance/)).

I am not sure this is what we want for our classes, as maybe is better to specify the class to be called for `normalize`; otherwise the logic can become very complicated to track down.

We can discuss this this Thu, or after the tutorial.

Reply:
ah ok, then I will just make a TODO for now, thanks